src/api/transactions.coffee

```diff
@@ -317,3 +393,28 @@ exports.removeTransaction = (transactionId) ->
     logger.info "User #{this.authenticated.email} removed transaction with id #{transactionId}"
   catch e
     utils.logAndSetResponse this, 'internal server error', "Could not remove transaction via the API: #{e}", 'error'
+
+
+
+###
+# construct transaction filtering dropdown options
+###
+exports.getTransactionsFilterOptions = ->
+
+  # Must be admin
+  if not authorisation.inGroup 'admin', this.authenticated
```

Comment:
non-admins may need this function to. You may need to search for only the channels and clients that a user is authorised to view.